dnn/data/converters/roco.py

```diff
@@ -55,35 +62,50 @@ def convert_annot(annot_path):
                float(bndbox.findtext('ymax')),
                float(bndbox.findtext('xmax'))]
         bbox.append(box)
+        diff = obj.findtext('difficulty')
+        diff = int(diff) if diff is not None else -1
+        difficulties.append(diff)
     objt = tf.io.serialize_tensor(tf.convert_to_tensor(objt, dtype=tf.int32))
     bbox = tf.io.serialize_tensor(tf.convert_to_tensor(bbox, dtype=tf.float32))
+    difficulties = tf.io.serialize_tensor(tf.convert_to_tensor(difficulties, dtype=tf.int8))
     # get encoded image binaries
-    image_path = annot_path.replace('image_annotation', 'image').replace('.xml', '.jpg')
+    image_path = annot_path.replace(
+        'image_annotation', 'image').replace('.xml', '.jpg')
     with open(image_path, 'rb') as f:
         image_target = f.read()
 
-    return image_target, objt.numpy(), bbox.numpy()
+    return image_target, objt.numpy(), bbox.numpy(), difficulties.numpy()
 
 
 def main(_argv):
     tf_set_memory_growth()
     os.makedirs(FLAGS.output, exist_ok=True)
     folder_ids = os.listdir(FLAGS.input)
     for n, folder_id in enumerate(folder_ids):
-        if not os.path.isdir(os.path.join(FLAGS.input, folder_id)): # skip irrelavent files
+        if not os.path.isdir(os.path.join(FLAGS.input, folder_id)):  # skip irrelavent files
             continue
-        output_path = os.path.join(FLAGS.output, 'ROCO_{}.tfrecords'.format(n + 1))
+        output_path = os.path.join(
+            FLAGS.output, 'ROCO_{}.tfrecords'.format(n + 1))
         if os.path.exists(output_path):
             os.remove(output_path)
         annot_dir = os.path.join(FLAGS.input, folder_id, 'image_annotation')
         with tf.io.TFRecordWriter(output_path) as writer:
             for annot_file in os.listdir(annot_dir):
                 annot_path = os.path.join(annot_dir, annot_file)
-                image_target, object_target, bbox_target = convert_annot(annot_path)
+                image_target, object_target, bbox_target, difficulties = convert_annot(
+                    annot_path)
                 example = tf.train.Example(features=tf.train.Features(feature={
                     'image': bytes_feature(image_target),
                     'class_n': bytes_feature(object_target),
                     'bbox_yxyx_n4': bytes_feature(bbox_target),
+                    '''
+                    difficulty:
+                        -1: DNE
```

Comment:
Maybe it's worth updating the purdue converter by setting all difficulties to -1. But if we use it only for verification then probably just leave it as it is

Reply:
Added that.